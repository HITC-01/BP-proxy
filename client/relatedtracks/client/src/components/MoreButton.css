.moreButtonContainer {
  position: relative;
}

.moreButtonClicked {
  position: absolute;
  z-index: 1;
  background-color: #ffffff;
  border-radius: 5px;
  border: 1px solid #ff5500;
  width: 20px;
  height: 20px;
  top: 9px;
  left: 265px;
}

.moreButton {
  position: absolute;
  z-index: 1;
  background-color: #ffffff;
  border-radius: 5px;
  border: 1px solid #999;
  width: 20px;
  height: 20px;
  top: 9px;
  left: 265px;
}

.moreButtonImgClicked {
  margin: auto;
  color: #ff5500;
  position: absolute;
  top: 2px;
  left: 2px;
}

.moreButtonImg {
  margin: auto;
  color: #000;
  position: absolute;
  top: 2px;
  left: 2px;
}

.moreButton:hover {
  border: 1px solid rgb(58, 57, 57);
}
