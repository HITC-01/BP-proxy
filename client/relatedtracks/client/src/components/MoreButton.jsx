import React from 'react';
import MoreModal from './MoreModal';
import styles from './MoreButton.css';

class MoreButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isClicked: false,
    };
    this.isClicked = this.isClicked.bind(this);
  }

  isClicked() {
    const { isClicked } = this.state;
    this.setState({ isClicked: !isClicked });
  }

  render() {
    const { isClicked } = this.state;
    return (
      <div className={styles.moreButtonContainer}>
        <div
          title="more"
          className={isClicked ? styles.moreButtonClicked : styles.moreButton}
          onClick={this.isClicked}
          onKeyDown={this.isClicked}
          role="presentation"
        >
          <MoreModal currentState={isClicked} />
          <i className={'fas fa-ellipsis-h '.concat(isClicked ? styles.moreButtonImgClicked : styles.moreButtonImg)} />
        </div>
      </div>
    );
  }
}

export default MoreButton;
