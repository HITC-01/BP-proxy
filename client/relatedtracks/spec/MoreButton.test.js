import React from 'react';
import renderer from 'react-test-renderer';
import MoreButton from '../client/src/components/MoreButton';

describe('tests the more button component', () => {
  test('Should render correctly', () => {
    const component = renderer.create(<MoreButton />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
