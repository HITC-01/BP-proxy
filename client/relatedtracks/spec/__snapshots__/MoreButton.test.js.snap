// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the more button component Should render correctly 1`] = `
<div
  className="moreButtonContainer"
>
  <div
    className="moreButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    title="more"
  >
    <div
      className="moreModal"
    >
      <div
        className="moreModal-contentOff"
      >
        <div
          className="moreModalData moreModalHover"
          title="Repost"
        >
          <span
            className="marginRight"
          >
            <i
              className="fas fa-exchange-alt moreButtonImg"
            />
          </span>
          <span>
            Repost
          </span>
        </div>
        <div
          className="moreModalData moreModalHover"
          title="Share"
        >
          <span
            className="marginRight"
          >
            <i
              className="fas fa-share moreButtonImg"
            />
          </span>
          <span>
            Share
          </span>
        </div>
        <div
          className="moreModalData moreModalHover"
          title="Add to Next up"
        >
          <span
            className="marginRight"
          >
            <i
              className="fas fa-sliders-h moreButtonImg"
            />
          </span>
          <span>
            Add to Next up
          </span>
        </div>
        <div
          className="moreModalData moreModalHover"
        >
          <span
            className="marginRight"
          >
            <i
              className="fas fa-chart-line moreButtonImg"
            />
          </span>
          <span>
            Add to playlist
          </span>
        </div>
        <div
          className="moreModalData moreModalHover"
          title="Go to station"
        >
          <span
            className="marginRight"
          >
            <i
              className="far fa-dot-circle moreButtonImg"
            />
          </span>
          <span>
            Station
          </span>
        </div>
      </div>
    </div>
    <i
      className="fas fa-ellipsis-h moreButtonImg"
    />
  </div>
</div>
`;
